src/Umbraco.Cms.Integrations.OAuthProxy/Controllers/OAuthProxyController.cs

```diff
@@ -62,6 +65,22 @@ public async Task ProxyTokenRequest()
             await Response.WriteAsync(content);
         }
 
+        private HttpClient GetClient(string serviceName)
+        {
+            var httpClient = _httpClientFactory.CreateClient($"{serviceName}Token");
+
+            var serviceAddressReplaceHeader = Request.Headers.FirstOrDefault(p => p.Key.Contains(ServiceAddressReplacePrefixHeaderKey));
```

Comment:
Would just suggest adding the note you had in the PR as a comment in the code for future reference.

```suggestion
            // Shopify's endpoint (and potentially others in future) for retrieving the access token is directly coupled with the shop's name.
            // As a result the address of the client needs to be updated with the request header value for the name of the shop.
            var serviceAddressReplaceHeader = Request.Headers.FirstOrDefault(p => p.Key.Contains(ServiceAddressReplacePrefixHeaderKey));
```